src/dataflow-types/src/reconciliation/command.rs

```diff
@@ -0,0 +1,528 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+
+//! Functionality to reconcile commands between a COMPUTE controller and a COMPUTE instance.
+//!
+//! The [ComputeCommandReconcile] and [StorageCommandReconcile] structs implement [GenericClient]
+//! the controller to reconnect after restarts. It maintains enough state to
+//! get a newly connected instance up-to-date and matches existing installed objects with
+//! what the controller wants to provide.
+//!
+//! [ComputeCommandReconcile] is designed to live in a COMPUTE instance and liberally uses `assert` to
+//! validate the correctness of commands. It is not intended to be part of a COMPUTE coordinator as
```

Comment:
I think everywhere you have "COMPUTE coordinator" we're using "COMPUTE controller".